sdk/armcore/poller.go

```diff
@@ -0,0 +1,1068 @@
+		if err := pt.initializeState(); err != nil {
+			return pt, err
+		}
+		// this initializes the polling header values, we do this during creation in case the
+		// initial response send us invalid values; this way the API call will return a non-nil
+		// error (not doing this means the error shows up in Future.Done)
+		return pt, pt.updatePollingMethod()
+	default:
+		return nil, fmt.Errorf("unsupported HTTP method %s", resp.Request.Method)
+	}
+}
+
+// ResumePollingTracker creates a polling tracker from a resume token string.
+// NOTE: this is only meant for internal use in generated code.
+func ResumePollingTracker(pollerType string, token string, errorHandler methodErrorHandler) (Poller, error) {
+	// unmarshal into JSON object to determine the tracker type
+	obj := map[string]interface{}{}
+	err := json.Unmarshal([]byte(token), &obj)
+	if err != nil {
+		return nil, err
+	}
+	if obj["pollerType"] != pollerType {
+		return nil, fmt.Errorf("cannot resume from this poller type")
+	}
+	if obj["method"] == nil {
+		return nil, fmt.Errorf("token is missing 'method' property")
+	}
+	method := obj["method"].(string)
+	switch strings.ToUpper(method) {
+	case http.MethodDelete:
+		pt := &pollingTrackerDelete{pollingTrackerBase: pollingTrackerBase{errorHandler: errorHandler}}
+		// now unmarshal into the tracker
+		err = json.Unmarshal([]byte(token), &pt)
+		if err != nil {
+			return nil, err
+		}
+		return pt, nil
+	case http.MethodPatch:
+		pt := &pollingTrackerPatch{pollingTrackerBase: pollingTrackerBase{errorHandler: errorHandler}}
+		// now unmarshal into the tracker
+		err = json.Unmarshal([]byte(token), &pt)
+		if err != nil {
+			return nil, err
+		}
+		return pt, nil
+	case http.MethodPost:
+		pt := &pollingTrackerPost{pollingTrackerBase: pollingTrackerBase{errorHandler: errorHandler}}
+		// now unmarshal into the tracker
+		err = json.Unmarshal([]byte(token), &pt)
+		if err != nil {
+			return nil, err
+		}
+		return pt, nil
+	case http.MethodPut:
+		pt := &pollingTrackerPut{pollingTrackerBase: pollingTrackerBase{errorHandler: errorHandler}}
+		// now unmarshal into the tracker
+		err = json.Unmarshal([]byte(token), &pt)
+		if err != nil {
+			return nil, err
+		}
+		return pt, nil
+	default:
+		return nil, fmt.Errorf("unsupported method '%s'", method)
+	}
+}
+
+// Poller defines the methods that will be called internally in the generated code for long-running operations.
+// NOTE: this is only meant for internal use in generated code.
+type Poller interface {
+	// Done signals if the polling operation has reached a terminal state.
+	Done() bool
+	// Poll sends a polling request to the service endpoint and returns the http.Response received from the endpoint or an error.
+	Poll(ctx context.Context, p azcore.Pipeline) (*http.Response, error)
+	// FinalResponse will perform a final GET and return the final response for the polling operation.
```

Comment:
Comment needs slight rewording to accommodate the `respType` param.

Reply:
Done